app/services/trading_strategies.py

```
@@ -4215,6 +4614,304 @@ async def _estimate_daily_volatility(self, symbol: str) -> float:
+    def _calculate_margin_efficiency(self, margin_by_exchange: Dict) -> float:
+        """Calculate margin usage efficiency across exchanges."""
+        try:
+            total_available = sum(info.get('available_margin', 0) for info in margin_by_exchange.values())
+            total_used = sum(info.get('used_margin', 0) for info in margin_by_exchange.values())
+            
+            if total_available == 0:
+                return 0.0
+                
+            efficiency = total_used / total_available
+            return max(0.0, min(efficiency, 1.0))
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)